src/java.base/share/classes/java/lang/StringUTF16.java

```diff
@@ -1314,9 +1315,12 @@ static Stream<String> lines(byte[] value) {
     }
 
     private static void putChars(byte[] val, int index, char[] str, int off, int end) {
```

Comment:
I recommend renaming this to `putCharsUnchecked` to indicate this has no bound checks.